### DISCUSSION TITLE
How to start application without show RootView?

### DISCUSSION THREAD

**Discussion by GF-Huang** [canton7/Stylet#209] [Q&A]
My application has a notify icon(tray icon) at the desktop right bottom. I want to startup my app without show any windows, until users to click the notify icon. How can I do that?

**Comment by mwdpb** [comment]
I had a similar but not exactly the same issue as yours.  See my thread for solutions.  You might need create your own bootstrapper class.

**Reply by canton7** [reply]
Just override Launch, and don't call anything, from your Bootstrapper

**Reply by mwdpb** [reply]
That'll work.  But you will need feed a dummy viewmodel to Bootstrapper which will drive OCD people like me crazy, :)

**Reply by mwdpb** [reply]
BTW, I got a question on your Bootstrapper implementation: can you pass TRootViewModel as a parameter to the constructor instead of using generic?  That way users have the freedom to not use root view at all.

**Reply by canton7** [reply]
You could, but I copied C.M back in the day, and I can't change it without it being a breaking change :)

**Reply by mwdpb** [reply]
Gotcha.  BTW, I've submitted a PR but you haven't approved it yet, was there any concerns on adding a layer in bootstrapping?

**Reply by canton7** [reply]
I left a comment which you never answered

**Reply by mwdpb** [reply]
Oops, my bad, I never realized there was a comment left there.  I'll stop here and move on to the PR thread.
